### PR TITLE
Enable OpenAPI documentation

### PR DESCRIPTION
**Pull request by rahulshamkuwar**
Added OpenAPI support.
Enabled Swagger CLI Plugin.
Added auto generation of DTO's from Prisma schema.
Known issue: cannot write descriptions for all DTO's since the package for generating the DTO's does not support it yet.

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -2,13 +2,16 @@ import { Controller, Request, Post, Body } from "@nestjs/common";
...
+import { CreateUserDto } from "src/database/generated/user";
 import { LoginUserDto } from "./dto/login_user.dto";
-import { RegisterUserDto } from "./dto/register_user.dto";
 
 @Controller("auth")
 export class AuthController {
   constructor(private authService: AuthService) {}
 
+  /**
+   * Login a user.
```
Comment (sspathare97) [inline src/auth/auth.controller.ts:13]: I have fallen so low, nitting for a comment. But I can't stop myself.
nit: Log in a user

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "athena-backend",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Comment (sspathare97) [inline package.json:3]: Why?

File: .env.example [hunk 3]
```diff
@@ -2,7 +2,7 @@ POSTGRES_DB=""
 POSTGRES_USER=""
 POSTGRES_PASSWORD=""
 
-DATABASE_URL="postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@localhost:5432/${POSTGRES_DB}?schema=public"
+DATABASE_URL="postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?schema=public"
```
Comment (sspathare97) [inline .env.example:5]: Why

**Review by sspathare97 — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (sspathare97) [reply package.json:3]: Well if 0.0.0 was the first version and 0.0.1 was the latest patch then adding this documentation which doesn't change any functionality is indeed a patch and should be 0.0.2

Comment (rahulshamkuwar) [reply package.json:3]: Oh my assumption was that the version starts at 0.1.0, not 0.0.0... in that case I can change it to 0.0.2.

**Review by sspathare97 — APPROVED** [review]

File: .env.example (hunk 3 above)
Comment (rahulshamkuwar) [reply .env.example:5]: Oh uh.............................. yeah this was wrong :sweat_smile:

**Pull request merged**